### ISSUE TITLE
change clc to home

### ISSUE THREAD

**Issue by wcollings** [wcollings/clc.github.io#13]
In the top links of every page, change the first link text from "Celtic Lyrics Corner" to "Home". Not that important, but seems like a good small change

**Comment by danmolloy** [comment]
Hi wcollings, I'd like to please do this as a first issue.

**Comment by danmolloy** [comment]
You've quite a few pages! Should be done by tomorrow.

**Comment by danmolloy** [comment]
Hi wcollings, you've really got a lot of pages!

I wonder if a more meaningful development could be to make a template which each page extends?
That way only one page would need to be updated in future changes.

**Comment by wcollings** [comment]
I have considered this, but there's a couple problems that would need to be solved along side that. First of all, github pages doesn't allow any backend stuff, so everything needs to be done client-side (unless I move to paying to have it hosted elsewhere). Secondly, I'm not normally a website guy (as can be seen by my repos), so my only thought was having all of that saved in a json file and then loading that, if I were to templatize everything. And I'm concerned about load times with that. I do agree that templatizing would be a major upgrade though. 

If you have a way to do that that's not "load every song in the database at first access" then I'd love to hear it! Like I said, this is really my first foray into anything above local python, and I'm primarily a C++ kinda guy, so I certainly don't know if there's anything better out there

**Comment by wcollings** [comment]
I have been scripting the shit out of everything using python and BeautifulSoup, but I agree that that's not necessarily the most efficient way to change things as uniform as this

**Comment by danmolloy** [comment]
No problem, I'll give it some thought and see if I can help you come up with a suitable solution.

**Comment by danmolloy** [comment]
Hi wcollings, this is much bigger than anything I've done but I could try saving them all in json files (as you mentioned) and render them using [NextJS](https://nextjs.org/). I think it should be fairly quick compared to just HTML, but again, this is bigger than anything I've done before. 
You should be able to keep the code here and deploy the site via [Vercel](https://vercel.com) for free.

If this is something you're interested I'm happy to give it a go.

**Comment by wcollings** [comment]
I'd say go for it, if you think you have a solution. Nothing's going to break if it doesn't work.

Moving across the country and starting a new job got in the way of this for a second but I should be back working on things again now

**Comment by danmolloy** [comment]
Hi wcollings, 
I'm sorry for the delay, I should be able to show you something in the next few days. 
Best, Dan

**Comment by wcollings** [comment]
Dan,
I'm making a new branch called "dynamic", I think I found a way to do this. Starting small with just the artist page for now. It turns out you can tell JS to load specific files dynamically, rather than just specifying them in the header like I (*stupidly*) thought you had to do. I've added a URL parameter for the artist's folder and then I just load that. Just need to generate a bunch of JSON files with the info contained in those artist files

**Comment by danmolloy** [comment]
Hi wcollings,
Sorry to be out of touch and string you along. 

I had a look at the dynamic branch - I think JSON files are definitely the way forward. I gave some thought to the most straightforward way to render the json and ended up making [a repo](https://github.com/danmolloy/clc-frontend) which I was hoping you could have a look at.  I've made a github pages for it, but at this point I think you'll need to run it on localhost for the json to render properly.

It's a template running React (basically HTML written in JS). 
With very limited CSS, I've made a navbar and a Homepage, artists list, single artist page.
You can find those respective JS pages in src/components.

I've also added the JSON files you created, although merged Altan and Breabach into a single file. You can find that in the public/data.json.

If you have a look at the artists page, you'll find that src/components/Artists.js imports the json file and creates a list of the artists, rendering the links. Using this same idea, I could also create a single artist page which will show the remaining json info.

I've also used React-Router to create the url paths. Again, it's using the json info to create the parameters to the single artist page. 

If this is something you're interested in, I'm more than happy to continuing working on it. I'm very happy to help and again I'm very sorry for the slow responses! Dan

**Comment by danmolloy** [comment]
Hi wcollings,

Adding on to the above - I made a [live example](https://danmolloy.github.io/clc-frontend/) on Github Pages. I've added the single artist page and formatted the album list from the json.

**Comment by wcollings** [comment]
Wow, ok. That's pretty sick. I'm a little bit concerned about merging artist files like that for the future, seems like it could bloat pretty fast and I want this to be as fast as possible. But I have no complaints otherwise! I can start working on extracting all that info into json files from what we already have though

**Comment by danmolloy** [comment]
Yes I agree. You could still have separate files i.e. artists beginning with A, B, C etc. 
As far as the rendering is concerned, it won't be an issue having multiple files.

I should say that my strengths lie more in front end rendering, not so much the storage of data. It could be worthwhile creating an issue to ask for opinions of the best way to store this data?

Also I wonder if someone may know a way to scrape the information from the html pages? It's possible someone could save you a lot of time!

Best, Dan

**Comment by wcollings** [comment]
Before you got here, the majority of this project was parsing and editing the HTML en masse with scripts, so extracting the information from them shouldn't be any harder than that. There's no way I was going to do that by hand in the first place! I'm working on that on my local repo and I'll push it before too long here, hopefully.

My background is more in lower-level stuff anyways, and data storage is certainly not alien to me, so I should have that end of things covered. I'll still make an issue for discussion, but as for carrying it out that's not a problem.
